oadr2/poll.py

```diff
@@ -75,23 +77,42 @@ def __init__(self, event_config, vtn_base_uri,
         self.__password = password
 
         self.poll_thread = None
-        start_thread = bool(start_thread)
-        self._init_client(start_thread)
+        if start_thread:  # this is left for backward compatibility
+            self.start()
 
-        logging.info(" +++++++++++++++ OADR2 module started ++++++++++++++ ")
+        logging.info(" +++++++++++++++ OADR2 module started ++++++++++++++ !!")
 
-    def _init_client(self, start_thread):
+    def start(self):
         '''
         Initialize the HTTP client.
 
         start_thread -- To start the polling thread or not.
         '''
+
+        if self.poll_thread and self.poll_thread.is_alive():
+            logging.warning("Thread is already running")
+            return
+
         self.poll_thread = threading.Thread(
             name='oadr2.poll',
             target=self.poll_vtn_loop)
         self.poll_thread.daemon = True
-        if start_thread:
-            self.poll_thread.start()
+        self._exit.clear()
+
+        self.poll_thread.start()
+        logging.info("Polling thread started")
+
+    def stop(self):
+        '''
+        Stops polling without stopping event controller
+        :return:
+        '''
+        if self.poll_thread is not None:
+            self.poll_thread.join(2)  # they are daemons.
```

Comment:
What should happen if this times out? Also, pass this as `timeout=2` for readability.

Reply:
This is a method used by original authors to stop threads. It was just split here to only stop polling or control threads

Reply:
:ok_hand: